### PR TITLE
Color difference: DeltaE cmc 0

### PR DESCRIPTION
**Pull request by arokem**
Addresses #1908?

### CODE REVIEW

**Comment by stefanv** [comment]
What happens in cases where things are *almost* equal, as you mentioned in #1908?

**Comment by soupault** [comment]
@stefanv difference of `1e-8` in any coordinate doesn't cause an error.

**Comment by arokem** [comment]
But sure enough, if: 

    lab2 = np.array([0. + np.finfo(float).eps, 1.59607713, 0.87755709])

Then it fails... Any ideas? 

Doing this: 

    if np.all(np.abs(lab1-lab2) < np.finfo(float).eps)

doesn't do the trick

**Comment by soupault** [comment]
@arokem `np.allclose`?

**Comment by arokem** [comment]
Yup. Thanks!

File: skimage/color/tests/test_delta_e.py [hunk 1]
```diff
@@ -137,6 +138,20 @@ def test_cmc():
 
     assert_allclose(dE2, oracle, rtol=1.e-8)
 
+    # Test that equal colors have a distance of 0:
+    lab1 = lab2
+    assert_equal(deltaE_cmc(lab1, lab2), np.zeros(oracle.shape))
+    # If they're *almost* equal, than that's still fine:
```
Comment (ahojnnes) [inline skimage/color/tests/test_delta_e.py]: than -> then

File: skimage/color/tests/test_delta_e.py [hunk 2]
```diff
@@ -137,6 +138,20 @@ def test_cmc():
...
+    lab2[0, 0] = lab2[0, 0] + np.finfo(float).eps
+    assert_almost_equal(deltaE_cmc(lab1, lab2), np.zeros(oracle.shape))
+
+    # Generalize to the single item case:
+    lab1 = lab2 = np.array([0., 1.59607713, 0.87755709])
+    assert_equal(deltaE_cmc(lab1, lab2), 0)
+
+    #lab2 = np.array([0.+ np.finfo(float).eps, 1.59607713, 0.87755709])
```
Comment (ahojnnes) [inline skimage/color/tests/test_delta_e.py]: Is it intended that this is commented? If yes, then add a comment why. If it is not needed any more, then remove the two lines. Thanks!

**Comment by stefanv** [comment]
Just the final few comments to address, then we're good to go.

**Comment by soupault** [comment]
@arokem hi! do you have time to finish this?

**Pull request closed without merging**